### PR TITLE
Support custom targets in WireRun and the Gradle plugin

### PR DESCRIPTION
**Pull request by lkerford**

### CODE REVIEW

**Comment by jrodbx** [comment]
I'm likely missing some context, but why do we want this?

**Comment by swankjesse** [comment]
I wanna replace our internal RPC service generators with the Wire gradle plugin. But I don’t want a hard dependency from Wire on Square’s internal RPC stuff.

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt [hunk 1]
```diff
@@ -41,11 +42,12 @@ abstract class WireOutput {
...
 
   /** Configure the project's graph to compile the files emitted. */
-  abstract fun applyToProject(
+  open fun applyToProject(
     project: Project,
     wireTask: TaskProvider<WireTask>?,
     kotlin: Boolean
-  )
+  ) {
+  }
```
Comment (oldergod) [inline wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt]: I wouldn't mind keep that as abstract, and let the implementer no-op it.

**Comment by oldergod** [comment]
CI is sad because it cannot find `Path#writeString`. Are we using a different/wrong java version on CI?

File: wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt (hunk 1 above)
Comment (lkerford) [reply wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WireOutput.kt]: fixed

**Comment by lkerford** [comment]
> CI is sad because it cannot find `Path#writeString`. Are we using a different/wrong java version on CI?

has been addressed

**Pull request merged**